snapcraft/internal/sources.py

```diff
@@ -307,10 +371,46 @@ def __init__(self, source, source_dir, source_tag=None,
         elif source_branch:
             raise IncompatibleOptionsError(
                 'can\'t specify a source-branch for a zip source')
-
-    def provision(self, dst, clean_target=True, keep_zip=False):
+        elif source_checksum:
+            raise IncompatibleOptionsError(
+                'can\'t specify source-checksum for a zip source right now')
+
+    def pre_check_checksum(self, source_checksum, zip):
```

Comment:
this looks like duplicate code, please consider adding it to FileBase (which also needs its __init__ updated)